mne_bids/utils.py

```diff
@@ -0,0 +1,212 @@
+    filename : str
+        The BIDS filename you wish to create.
+
+    Examples
+    --------
+    >>> print(make_bids_filename(subject='test', session='two', task='mytask', suffix='data.csv')) # noqa
+    sub-test_ses-two_task-mytask_data.csv
+    """
+    order = OrderedDict([('sub', subject),
+                         ('ses', session),
+                         ('task', task),
+                         ('acq', acquisition),
+                         ('run', run),
+                         ('proc', processing),
+                         ('recording', recording)])
+    if order['run'] is not None and not isinstance(order['run'], str):
+        # Ensure that run is a string
+        order['run'] = '{:02}'.format(order['run'])
+
+    _check_types(order.values())
+
+    if not any(isinstance(ii, str) for ii in order.keys()):
+        raise ValueError("At least one parameter must be given.")
+    filename = []
+    for key, val in order.items():
+        if val is not None:
+            filename.append('%s-%s' % (key, val))
+    if isinstance(suffix, str):
+        filename.append(suffix)
+    filename = '_'.join(filename)
+    if isinstance(prefix, str):
+        filename = os.path.join(prefix, filename)
+    return filename
+
+
+def make_bids_folders(subject, session=None, kind=None, root=None,
+                      make_dir=True):
+    """Create a BIDS folder hierarchy.
+
+    This creates a hierarchy of folders *within* a BIDS dataset. You should
+    plan to create these folders *inside* the root folder of the dataset.
+
+    Parameters
+    ----------
+    subject : str
+        The subject ID. Corresponds to "sub".
+    kind : str
+        The kind of folder being created at the end of the hierarchy. E.g.,
+        "anat", "func", etc.
+    session : str | None
+        The session for a item. Corresponds to "ses".
+    root : str | None
+        The root for the folders to be created. If None, folders will be
+        created in the current working directory.
+    make_dir : bool
+        Whether to actually create the folders specified. If False, only a
+        path will be generated but no folders will be created.
+
+    Returns
+    -------
+    path : str
+        The (relative) path to the folder that was created.
+
+    Examples
+    --------
+    >>> print(make_bids_folders('sub_01', session='my_session',
+                                kind='meg', root='path/to/project', make_dir=False))  # noqa
+    path/to/project/sub-sub_01/ses-my_session/meg
+    """
+    _check_types((subject, kind, session, root))
+
+    path = ['sub-%s' % subject]
+    if isinstance(session, str):
+        path.append('ses-%s' % session)
+    if isinstance(kind, str):
+        path.append(kind)
+    path = os.path.join(*path)
+    if isinstance(root, str):
+        path = os.path.join(root, path)
+
+    if make_dir is True:
+        _mkdir_p(path)
+    return path
+
+
+def make_dataset_description(path, name=None, bids_version=None, license=None,
```

Comment:
we should just define `bids_version` somewhere and make sure we produce files compatible with that BIDS version. It will be a mess to support different versions of BIDS